native-audio/app/src/main/java/com/example/nativeaudio/NativeAudio.java

```diff
@@ -308,6 +336,34 @@ protected void onDestroy()
         shutdown();
         super.onDestroy();
     }
+    private void requestResourcePermission() {
+        if (ActivityCompat.checkSelfPermission(this, Manifest.permission.RECORD_AUDIO) !=
+                PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this,
+                    new String[]{Manifest.permission.RECORD_AUDIO },
+                    NATIVE_AUDIO_REQUEST);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
+                                           @NonNull int[] grantResults) {
+
+        /*
+         * if any permission failed, the sample could not play
+         */
+        if (NATIVE_AUDIO_REQUEST != requestCode) {
+            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+            return;
+        }
+
+        for (int res:grantResults) {
+            if (res != PackageManager.PERMISSION_GRANTED) {
```

Comment:
How are you handling the permission denied option here? It looks like none of the buttons will function without the permission and there is no user feedback that this is the reason.
What about bringing up a dialog after denial from which the user can re-request the permission or close the app?